test/y2storage/blk_device_test.rb

```diff
@@ -1426,4 +1451,42 @@ def create_encrypted_partition(disk, slot_index)
       expect(subject.windows_suitable?).to eq(false)
     end
   end
+
+  describe "#preferred_mount_by" do
+    let(:device_name) { "/dev/sda1" }
+
+    before do
+      allow(device).to receive(:udev_paths).and_return ["udev_path"]
+    end
+
+    let(:by_device) { Y2Storage::Filesystems::MountByType::DEVICE }
+    let(:by_path) { Y2Storage::Filesystems::MountByType::PATH }
+    let(:all_suitable) { [by_device, by_path] }
+
+    it "returns the best mount by from all the suitable ones" do
```

Comment:
To be honest, I would rather prefer "mount-by" (or even *"mount by"*) instead of "mount by" ("by from" looks wrong). However, I know that we use that expression in many places and perhaps it is not worth changing it now.

Reply:
I fully agree, I just tried to respect the original ones written by @joseivanlopez